### ISSUE TITLE
Not playing next item

### ISSUE THREAD

**Issue by prydda** [im85288/service.upnext#170]
Using Embycon to playback and embuary skin. 
I'm getting the popup to play next episode but it never does, it jumps back to current season. 

logfile: http://ix.io/2bxL

**Comment by dagwieers** [comment]
The log is mostly empty. Did you enable debug logging?

**Comment by prydda** [comment]
Nope, sorry. 
here is with debug activated: 
https://del.dog/ygarfullyg

**Comment by dagwieers** [comment]
I don't know who wrote this dogbin service, but it is unusable in Chrome. Is it mining bitcoins perhaps?

How hard is it to create a service to show a text file in a browser without bringing the browser to a grinding halt. Sorry, try again.

**Comment by dagwieers** [comment]
Maybe you can attached it to this issue instead?

**Comment by prydda** [comment]
Not sure, used it for android logs. ix.io didn't work - I beileve the file was to large? (5.4mb) 

here is a github gist: https://gist.github.com/prydda/4cd7c341cb0d235a8a0f49c621525435

**Comment by dagwieers** [comment]
You can also attach it, see below: *Attach files by dragging & dropping, selecting or pasting them.*

**Comment by dagwieers** [comment]
The only logged information from Up Next is receiving the information:
> DEBUG: [service.upnext] Api -> addon_data_received called with data {u'play_info': {u'item_id': u'539893', u'auto_resume': False, u'force_transcode': False}, u'next_episode': {u'rating': 0.0, u'tvshowid': u'The Sunny Side Up Show', u'art': {u'tvshow.fanart': u'', u'thumb': u'https://emby.popcorney.xyz:443/emby/Items/539893/Images/Primary/0?Format=original&Tag=a0d62349704fd8a9c6de672f2236f06e|verifypeer=false', u'tvshow.clearart': u'https://emby.popcorney.xyz:443/emby/Items/539776/Images/Art/0?Format=original&Tag=|verifypeer=false', u'tvshow.landscape': u'', u'tvshow.poster': u'https://emby.popcorney.xyz:443/emby/Items/539776/Images/Primary/0?Format=original&Tag=6799c5baaaba14cd1567c7e1a8885320|verifypeer=false', u'tvshow.clearlogo': u''}, u'plot': None, u'season': 3, u'title': u'Vattnet g\xe5r', u'episodeid': u'539893', u'firstaired': None, u'episode': 10, u'playcount': 0, u'showtitle': u'The Sunny Side Up Show'}, 'id': u'embycon_play_action', u'current_episode': {u'rating': 0.0, u'tvshowid': u'The Sunny Side Up Show', u'art': {u'tvshow.fanart': u'', u'thumb': u'https://emby.popcorney.xyz:443/emby/Items/539892/Images/Primary/0?Format=original&Tag=b02a4c8f959d4fe78cf4ef0dd80db1d7|verifypeer=false', u'tvshow.clearart': u'https://emby.popcorney.xyz:443/emby/Items/539776/Images/Art/0?Format=original&Tag=|verifypeer=false', u'tvshow.landscape': u'', u'tvshow.poster': u'https://emby.popcorney.xyz:443/emby/Items/539776/Images/Primary/0?Format=original&Tag=6799c5baaaba14cd1567c7e1a8885320|verifypeer=false', u'tvshow.clearlogo': u''}, u'plot': u'DirectStream\nI Saltsj\xf6baden \xf6ppnar en kemtv\xe4tt som b\xe4r samma namn som Mickans och Freddes efternamn. N\xe4r Anna och Alex unders\xf6ker olika f\xf6rlossningsalternativ anklagar Anna Alex f\xf6r att vara en morsgris. Han f\xf6rs\xf6ker bevisa motsatsen men lyckas inte n\xe5got vidare.', u'season': 3, u'title': u'Kemtv\xe4tten', u'episodeid': u'539892', u'firstaired': 2012, u'episode': 9, u'playcount': 0, u'showtitle': u'The Sunny Side Up Show'}}

Which looks fine, but I see nothing else in the logs. Not even the notification itself, which is weird. I think we need to look at improving the default logging when debug logging is enabled. (I also see from the debug logs we can improve how we read the settings, this is suboptimal)

**Comment by prydda** [comment]
[kodi.log](https://github.com/im85288/service.upnext/files/4198478/kodi.log)

I forgot put logging on from Up Next addon in the last log. Here is one with the same issue and debug on from Up Next - Settings - Expert - LogLevel Debug

**Comment by dagwieers** [comment]
Ok, so this Kodi log includes this line:
```
2020-02-13 14:12:35.695 T:3726914432   DEBUG: [service.upnext] Api -> Sending base64 data to add-on to play: {u'item_id': u'539896', u'auto_resume': False, u'force_transcode': False}
```

Which instructs the calling add-on to start playing the next episode. So we have 2 possibilities:
- The data we send was incorrectly formatted (e.g. to work with Addon Signals)
- The data we send was not received by the add-on correctly or timely (which may indicate an issue with Addon Signals or the add-on itself)

Since we have unit tests for the former problem testing Addon Signals, I doubt this is the case. It would be best if the add-on devs could investigate what is going on in their add-on.
- Do they receive the info or not?
- Is the info correctly formatted? (we return what they send)
- Do they correctly and timely act on the response?

In v1.1.0 we added the possibility to have Up Next call the add-on directly to instruct it to play a specific episode, which is often more straightforward and easier. This would be something for the Embycon devs to consider.

**Comment by prydda** [comment]
@TeamB is this something for you?

**Comment by prydda** [comment]
@faush01 is there something you can do?

**Comment by dagwieers** [comment]
For completeness, we think it is better to not require Addon Signals, as it adds unneeded complexity if the add-on already has a Monitor or Player instance. And with the new addition of `play_url` the add-on doesn't need to have this complexity. See: https://github.com/im85288/service.upnext/pull/133

I apologize for not having better documentation, we desperately need a rewrite of the documentation for developers. Things have become a lot easier nowadays.

**Comment by faush01** [comment]
The signals data approach should still work.

**Comment by dagwieers** [comment]
Yes it should. But there are a few known issues in the latest release. Not sure if it could be related.
https://github.com/ruuk/script.module.addon.signals/pull/9

**Comment by prydda** [comment]
last night my other device (same OS - coreelec) and embycon plugin as playback content did actually play next episode from upnext addon as it should? - I will compare the settings and report back.

**Comment by dagwieers** [comment]
It would be nice to get to the bottom of this before we are releasing v1.1.1.
If it cannot be reproduced, I would rather close this issue until we can.

**Comment by dagwieers** [comment]
It is possible that #198 fixes also this issue. If this is easily reproduced at your end, please test out the fix in #198.
(You can download [this zip](https://github.com/MoojMidge/service.upnext/archive/master.zip) and install it using the ZIP file)

**Comment by dagwieers** [comment]
So, I am sorry, I have no idea what Emby or Embycon exactly is so have a hard time to relate to the issue at hand.
But could it be related to a connection limit in Emby (one connection) as reported in issue #201 ?

**Comment by dagwieers** [comment]
Please test the upcoming v1.1.3 release by testing PR #217 and reporting back.

**Comment by sasagr** [comment]
I think I have a similar (if not the same) issue.
Here the log. https://paste.kodi.tv/ewuveborig
There is no error but it tells me that he cannot play next item.
This is happening if I play the current episode from the Next Up section within the embystuary skin. If I play the episode from the season folder, it works fine.

**Comment by MoojMidge** [comment]
Your issue is seemingly unrelated. What you are likely seeing is this issue: https://github.com/im85288/service.upnext/issues/255

Has been fixed but a new release needs to be pushed out.

**Comment by sasagr** [comment]
> Your issue is seemingly unrelated. What you are likely seeing is this issue: #255
> 
> Has been fixed but a new release needs to be pushed out.

that's great news! but can I try it before it is pushed out? The link of the update you aded in the other issue, seems not be valid anymore

**Comment by MoojMidge** [comment]
You can download the latest code from this repository directly, but as you are using Kodi 19 you will either need to execute the makefile or modify the addon.xml file:
https://github.com/im85288/service.upnext/blob/b5df61fc187fe71c1bd0f437b7d3cf0a18adbb86/addon.xml#L4

So that it reads
`        <import addon="xbmc.python" version="3.0.0"/>`

**Comment by cgwprs** [comment]
I am having this issue "Not playing next item" with two addons, Seren and Asgard. Environment is Xubuntu 20.04, Kodi 19.5, current stable release of both addons. I will try to get a debug level log uploaded here today.

**Comment by MoojMidge** [comment]
This issue was about Embycon and, as far as I know, has been resolved.

If you are having issues with other video plugins, then a log is needed. That being said, I'm pretty sure none of the video plugins you mentioned support the use of Up Next, and instead handle playback of the next video themselves.
